psydac/linalg/solvers.py

```diff
@@ -62,27 +61,41 @@ def inverse(A, solver, **kwargs):
     # `InverseLinearOperator` subclass in this module:
     solvers_dict = {
         'cg'       : ConjugateGradient,
-        'pcg'      : PConjugateGradient,
         'bicg'     : BiConjugateGradient,
         'bicgstab' : BiConjugateGradientStabilized,
-        'pbicgstab': PBiConjugateGradientStabilized,
         'minres'   : MinimumResidual,
         'lsmr'     : LSMR,
         'gmres'    : GMRES,
     }
 
+    # Only these solvers accept a preconditioner argument for now:
+    solvers_with_pc = ['cg', 'bicgstab']
+
+    # Convert input solver string to lower case
+    solver = solver.lower()
+    
     # Check solver input
     if solver not in solvers_dict:
         raise ValueError(f"Required solver '{solver}' not understood.")
 
+    # If pc not accepted by solver: discard (if None) or raise an error
+    if 'pc' in kwargs and solver not in solvers_with_pc:
+        pc = kwargs.pop('pc')
+        if pc is not None:
+            raise ValueError(f"Invalid preconditioner '{pc}' passed with solver '{solver}'.")
+
     assert isinstance(A, LinearOperator)
 
+    # see failing tests in test_solvers.py
+    if A.dtype == complex and solver in ['lsmr', 'bicg']:
+        raise Warning(f"Solver '{solver}' currently not tested with complex operators.")
```

Comment:
Instead of raising the `Warning` exception, we should `import warnings` and then

```suggestion
    if A.dtype == complex and solver in ['lsmr', 'bicg']:
        msg = f"Solver '{solver}' currently not tested with complex operators."
        warnings.warn(msg, Warning)
```